packages/flutter_markdown/lib/src/builder.dart

```diff
@@ -686,6 +686,15 @@ class MarkdownBuilder implements md.NodeVisitor {
     }
   }
 
+  bool isTwoTextWidgetsSimilar(Widget current, Widget previous) {
+    if ((current is Text || current is RichText) &&
```

Comment:
I can't argue with your test results, but should this not be:

```
if ((current is Text && previous is Text) ||
        (current is RichText && previous is RichText)) {
```

Reply:
This is actually right, because we can combining two texts even if they are `RichText` or `Text`